### ISSUE TITLE
Enable pause status for the milestones

### ISSUE THREAD

**Issue by vikasrohit** [appirio-tech/connect-app#2968]
As of now we don't support pause status for a milestone which means that once a milestone is activated, it has to be completed whether on time or before/later of the scheduled time. There are use cases where we need to have a new status `paused` of the milestone which would be used when the milestone activity is blocked due to some reason which should not consider the time elapsed after pause action, in calculation of the total time taken to complete the milestone. 
It is yet to be decided if we want every user who can edit a milestone to be able to change the status to `paused`. We might want only admins to be able to do so to prevent misuse of this status to keep delivery in time.

**Comment by maxceem** [comment]
1. Would it be a button for phases or milestones? As we cannot have two milestones active at the same time looks like pausing a milestone means pausing a phase, so probably we can keep status options `Paused` active for phase after activating.

2. IMHO I would make it available for copilots to pause. As from the experience, it looks like the copilots are the most in the context of what is going technically and if we the timeline is met or no. Also, I think anyway copilots are responsible for meeting deadlines in front of clients. Making only admins who can pause could cause make this feature very hard to use, as every time has to reach for admins.

**Comment by vikasrohit** [comment]
1. I think for now lets add new button called `Pause Milestone` which would pause the milestone and at the same time it should also cause the phase to be paused as well. I guess we need to do the second part in backend service itself as cascade effect of the first action.

2. Agreed with all of your concerns, however, we can not let users to mis use this feature as well. So, do you think you have any way to prevent cases where a CA/copilot is pausing the milestone just for fitting the timeline in promised deadline? And btw, by admins, I mean contacting support team so that we can have valid audit record for all such actions. We are open for suggestions in this regard to make things easy for copilots.

**Comment by maxceem** [comment]
2. I think every time `copilot`/`CA` pause the milestone, we ask him to enter the reason and after display the reason so everybody admins and customers can see it. This way we will give the `copilot`/`CA` the flexibility to pause/resume milestones when they think it's necessary and the same time as everybody would see the reason it should prevent of abusing this feature. As we cannot put something like `we are waiting for customer` if the customer can see it and he knows that there are nothing to wait from him, for example.

**Comment by maxceem** [comment]
Note, currently, we can change the duration of each milestone without any confirmation.

**Comment by vikasrohit** [comment]
Good suggestion. Agreed, please keep a field for `reason` in the new table that we would need for keeping track of status changes of the phase statuses.

**Comment by maxceem** [comment]
@vikasrohit the server-side part is ready for this task and has been done via PR to project service https://github.com/topcoder-platform/tc-project-service/pull/322.

Should we implement some UI for it in the Connect App?

**Comment by vikasrohit** [comment]
@maxceem we have to implement this with new work management features.

**Comment by maxceem** [comment]
@vikasrohit ok, do we have some designs for new work management features?

**Comment by vikasrohit** [comment]
Yep, they are being worked on and should be ready very soon. We would need them implemented in 2.5 release.
